home_task_1/src/components/Functional/Search.js

```diff
@@ -0,0 +1,41 @@
+import { useState } from 'react';
+import { companies } from '../../data/companies';
+
+function FunctionalSearch() {
+  const [query, setQuery] = useState("");
+
+  const onChange = (e) => {
+    setQuery(e.target.value)
+  }
```

Comment:
see `useCallback` hook and update this function

Reply:
Modified